airbyte-commons-worker/src/main/java/io/airbyte/workers/process/DockerProcessFactory.java

```diff
@@ -169,6 +173,31 @@ public Process create(final String jobType,
     }
   }
 
+  /**
+   * !! ONLY FOR DEBUGGING, SHOULD NOT BE USED IN PRODUCTION !! If you set the DEBUG_CONTAINER_IMAGE
+   * environment variable, and it matches the image name of a spawned container, this method will add
+   * the necessary params to connect a debugger. For example, to enable this for
+   * `destination-bigquery` start the services locally with: ``` VERSION="dev"
+   * DEBUG_CONTAINER_IMAGE="destination-bigquery" docker compose -f docker-compose.yaml -f
+   * docker-compose.debug.yaml up ``` Additionally you may have to update the image version of your
+   * target image to 'dev' in the UI of your local airbyte platform. See the
+   * `docker-compose.debug.yaml` file for more context.
+   *
+   * @param containerName the name of the container which could be debugged.
+   * @return A list with debugging arguments or an empty list
+   */
+  static List<String> localDebuggingOptions(final String containerName) {
+    final boolean shouldAddDebuggerOptions =
+        Optional.ofNullable(System.getenv("DEBUG_CONTAINER_IMAGE")).filter(StringUtils::isNotEmpty)
+            .map(ProcessFactory.extractShortImageName(containerName)::equals).orElse(false)
+            && Optional.ofNullable(System.getenv("DEBUG_CONTAINER_JAVA_OPTS")).isPresent();
+    if (shouldAddDebuggerOptions) {
+      return List.of("-e", "JAVA_TOOL_OPTIONS=" + System.getenv("DEBUG_CONTAINER_JAVA_OPTS"), "-p5005:5005");
```

Comment:
Not entirely sure how this `JAVA_TOOL_OPTIONS=X` is getting rebuilt, do we need some extra quotes here?

Reply:
I think its ok as is, this is all getting passed into a command List/Array to be sent to the docker run command.  Specifically not using `JAVA_TOOL_OPTIONS` as the env variable so that the container which is running this doesn't try and process it, but the result of this command should effectively be a List with 3 strings